libdnf5/rpm/solv/goal_private.cpp

```diff
@@ -283,6 +288,16 @@ bool GoalPrivate::limit_installonly_packages(libdnf5::solv::IdQueue & job, Id ru
                 Id action = SOLVER_ERASE;
                 if (j < static_cast<int>(installonly_limit)) {
                     action = SOLVER_INSTALL;
+                } else {
+                    // We want to avoid reinstalling packages marked for ERASE, therefore
+                    // if identical unused provider is available we need to lock it.
+                    Solvable * s = spool.id2solvable(id);
+                    for (Id aup_id : available_unused_providers) {
+                        Solvable * aup_s = spool.id2solvable(aup_id);
+                        if (solvable_identical(s, aup_s) != 0) {
```

Comment:
I am not sure whether the test is required. I understand we want to avoid reinstall but what about the install of another version to prevent  removal of dependent package.

Reply:
But it is not wrong - the patch will prevents only reinstall and other cases are less likely.

Reply:
I have updated the PR to match the old libdnf PR: https://github.com/rpm-software-management/libdnf/pull/1625